### ISSUE TITLE
issue: useFieldArray root error is set to the actual field array root object, not the .root property using resolvers

### ISSUE THREAD

**Issue by Moshyfawn** [react-hook-form/react-hook-form#8759]
### Version Number

7.34.0

### Codesandbox/Expo snack

https://codesandbox.io/s/immutable-dawn-v1iigh?file=%2Fsrc%2FApp.tsx

### Steps to reproduce

1. Go to [csb](https://codesandbox.io/s/immutable-dawn-v1iigh?file=%2Fsrc%2FApp.tsx)
2.1 See the `errors` object shape in the console
```json
{
  "message": "The fields list is too small",
  "type": "too_small"
}
```
2.2 See the field array error message paragraph `<p>` content is empty
```jsx
{errors.fields && <p>{errors.fields.root?.message}</p>}
```

### Expected behaviour

The `errors` object shape is the same between using the **"native" `useFieldArray`** and **resolver** validation from [the `7.34.0` release](https://github.com/react-hook-form/react-hook-form/releases/tag/v7.34.0)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by Moshyfawn** [comment]
It might be more of a discussion than an issue, but I feel like we need to address the discrepancy between the API at least in the docs

**Comment by bluebill1049** [comment]
thanks, @Moshyfawn that's a resolver, I haven't considered supporting it just yet. maybe the next feature? Is this probably something to do in the resolver repo?

**Comment by Moshyfawn** [comment]
I wasn't quite sure if I should file this issue at react-hook-form/resolvers or even react-hook-form/documentation. Since this repo aggregates all the infra, I left it here :P

I feel like we should have some note somewhere, whether it's a note in docs for the resolvers or `formState` section or at least an issue ppl can search for know that we know lol _(hence the long issue title with a bunch of "keywords" to possible match someone's search query for this issue)_

**Comment by bluebill1049** [comment]
I just release I haven't updated the doc as well :( too busy... i will do it this weekend. I will ping you there and you can help me on the wording as well 🤗

**Comment by Moshyfawn** [comment]
> I just release I haven't updated the doc as well :( too busy... i will do it this weekend. I will ping you there and you can help me on the wording as well 🤗

Yep, sounds good! I'm currently swamped with a new release as well, so if you can ping me when you want to work on docs, I can for sure help you update the info!
